spring-cloud-gcp-pubsub/src/test/java/org/springframework/cloud/gcp/pubsub/core/PubSubTemplateTests.java

```diff
@@ -201,6 +202,14 @@ public void testSend_onFailure() {
 				.hasMessageContaining("future failed.");
 	}
 
+	@Test
+	public void testPublish_onFailureWithPayload() {
+		ListenableFuture<String> future = this.pubSubTemplate.publish("testTopic", this.pubsubMessage);
+		this.settableApiFuture.setException(new Exception("Publish failed"));
+		assertThatThrownBy(() -> future.get()).isInstanceOf(ExecutionException.class).hasMessageContaining("permanating")
```

Comment:
I think we need to check in the test that we have already that new `PubSubDeliveryException`  with expected `this.pubsubMessage` populated. I believe that was a main goal of this fix.